rfcs/0052-extensible-udt/0052-extensible-udt.md

```diff
@@ -0,0 +1,370 @@
+            owner_signature: <None>				
+            raw_extension_data: 
+                <vec> ScriptVec
+                    <script>
+                <...>
+            extension_data: 
+                <vec> BytesVec
+                    <data>
+                <...>
+```
+
+The only difference here is that `XudtWitnessInput` in `input_type` or
+`output_type` field in the corresponding WitnessArgs structure, contains raw
+extension data in `ScriptVec` data structure, xUDT script must first validate
+that the hash of raw extension data provide here, is the same as blake160 hash
+included in xUDT args. After this, it uses the same logic as the previous
+workflow.
+
+### Owner Mode Without Consuming Cell
+
+As described above, If an input cell uses an input lock script with same script
+hash as the owner lock script hash, the `is_owner_mode` will be set to true. It
+isn’t convenience: this requires extra cell to be consumed. With `owner_script`
+and `owner_signature` with proper values, we can use owner mode without extra
+cell.
+
+```yaml
+Inputs:
+    <vec>
+			<Any input cells>
+    <...>
+Outputs:
+    <vec> xUDT_Cell
+        Data:
+            <amount: uint128> <xUDT data>
+        Type:
+            code_hash: xUDT type script
+            args: <owner lock script hash 1> <xUDT args>
+        Lock:
+            <user defined>
+    <...>
+Witnesses:
+    WitnessArgs structure:
+        Lock: <user defined>
+        Input Type: <None>
+        Output Type: XudtWitnessInput
+            owner_script: <owner script 1>
+            owner_signature: <signature 1>				
+            raw_extension_data: 
+                <vec> ScriptVec
+                    <script>
+                <...>
+            extension_data: 
+                <vec> BytesVec
+                    <data>
+                <...>
+```
+
+The example above shows a scenario of owner mode without consuming cell. This
+transaction can be valid even without any input cell. The `<owner lock script
+hash 1>` is the same as blake160 hash of `<owner script 1>`. We can implement an
+extension script as `<owner script 1>` which has same functionality to
+[secp256k1/blake160 lock
+script](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0024-ckb-genesis-script-list/0024-ckb-genesis-script-list.md#secp256k1blake160).
+The `<signature 1>` can be used by `<owner script 1>` to place signature
+information.
+
+## Deployment
+
+An [implementation](https://github.com/nervosnetwork/ckb-production-scripts/blob/master/c/xudt_rce.c) of
+the spec above has been deployed to Mirana CKB mainnet and Pudge testnet:
+
+- Mirana(mainnet)
+
+| parameter | value |
+| --- | --- |
+| code_hash | 0x50bd8d6680b8b9cf98b73f3c08faf8b2a21914311954118ad6609be6e78a1b95 |
+| hash_type | data1 |
+| tx_hash | 0xc07844ce21b38e4b071dd0e1ee3b0e27afd8d7532491327f39b786343f558ab7 |
+| index | 0x0 |
+| dep_type | code |
+
+This script is not upgradeable due to zero lock (lock args with all zeros).
```

Comment:
When I see the word `zero lock`, I thought it is a lock whose `code_hash` is all zero, not just with all zeros lock args.

## [Some ckb-system-scripts](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0024-ckb-genesis-script-list/0024-ckb-genesis-script-list.md) such as SECP256K1 and Nervos DAO are using lock script
```json
{
  "code_hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "hash_type": "data",
  "args": "0x"
}
```

## While xUDT is using
```json
"a_lock_with_zero_args": {
  "code_hash": "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
  "hash_type": "type"
  "args": "0x0000000000000000000000000000000000000000",
},
```

I know it is almost impossible to unlock `a_lock_with_zero_args`.
Just got a little confused with the `zero lock` wording.

Reply:
It's limited by ckb-cli deployment tools. It can't set code_hash to all zero.